__tests__/utils/permission.test.js

```diff
@@ -0,0 +1,80 @@
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+  });
+  it('checkCameraPermission unavailable should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('unavailable');
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionUnavailable);
+  });
+  it('checkCameraPermission denied should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('denied');
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionDenied);
+  });
+
+  it('checkCameraPermission default should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('default');
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual('');
+  });
+});
+
+describe('requestExternalStoragePermission shpuld run correctly', () => {
+  beforeAll(() => {
+    jest.mock('react-native/Libraries/Utilities/Platform', () => ({
+      OS: 'android', // or 'ios'
+      select: () => null
+    }));
+  });
+  it('requestExternalStoragePermission granted should run correctly', async () => {
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue({
+      'android.permission.READ_EXTERNAL_STORAGE': 'granted',
+      'ios.permission.PHOTO_LIBRARY': 'granted',
+      'android.permission.READ_MEDIA_IMAGES': 'granted'
+    });
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+  });
+  it('requestExternalStoragePermission unavailable should run correctly', async () => {
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue({
+      'android.permission.READ_EXTERNAL_STORAGE': 'unavailable',
+      'ios.permission.PHOTO_LIBRARY': 'unavailable',
+      'android.permission.READ_MEDIA_IMAGES': 'unavailable'
+    });
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionUnavailable);
+  });
+  it('requestExternalStoragePermission denied should run correctly', async () => {
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue({
+      'android.permission.READ_EXTERNAL_STORAGE': 'denied',
+      'ios.permission.PHOTO_LIBRARY': 'denied',
+      'android.permission.READ_MEDIA_IMAGES': 'denied'
+    });
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionDenied);
+  });
+});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a mistake in the test cases for `requestExternalStoragePermission`. You are spying on and expecting `mock.check` to have been called, but it should be `mock.requestMultiple` since that's what you're using in these tests.
```diff
-    expect(mock.check).toHaveBeenCalled();
+    expect(mock.requestMultiple).toHaveBeenCalled();
```
Please make this change for lines 57, 67, and 77.


<!-- This is an auto-generated comment by OSS CodeRabbit -->